frontend/src/services/common/normalizerService.js

```diff
@@ -0,0 +1,17 @@
+export const normalizerService = {
+    normalize(response, shape = {}){
+        const { data } = response;
+
+        const transformed = data.map(item => {
```

Comment:
It won't work. First of all your result will be like this: 

```
[ {1: {id: 1}}, {2: {id: 2}} ]
```

but we expect:

```
{
   1: { id: 1 },
   2: { id: 2 }
}
```

And second, you can't rely on `data` because it can be an object, not only an array. My suggestion is to send into normalize method an object. And if we expect response.data as an array we will use reduce method and call normalize on each item. You can even create a few methods, which are like normilizeArray, normilizeObject, and do:

```
normilize(response, shape = {}) {
   let result = {}; 
   
    if (Array.isArray(response.data)) {
         result = this.normilizeArray(response.data, shape);
    } else {
         result = this.normilizeObject(response.data, shape);
    }

    return result;
}
```

It looks much clear, isn't it?